cmd/up.go

```diff
@@ -38,26 +41,57 @@ func Up() *cobra.Command {
 			fmt.Println("Activating your cloud native development environment...")
 
 			ctx, cancel := context.WithCancel(context.Background())
-			defer cancel()
 			var wg sync.WaitGroup
 			defer shutdown(cancel, &wg)
 
-			d, err := ExecuteUp(ctx, &wg, dev, namespace)
+			reconnectChannel := make(chan struct{}, 1)
+			runtime.ErrorHandlers = []func(error){
+				func(e error) {
+					if strings.Contains(e.Error(), "an error occurred forwarding") ||
+						strings.Contains(e.Error(), "error creating forwarding stream") ||
+						strings.Contains(e.Error(), "lost connection to pod") {
+						log.Debugf("connectivity error, sending a reconnection request: %s", e)
+						reconnectChannel <- struct{}{}
+						return
+					}
+
+					log.Debugf("unhandled error: %s", e)
+				},
+			}
+
+			d, err := ExecuteUp(ctx, &wg, dev, namespace, reconnectChannel)
 			if err != nil {
 				return err
 			}
 
-			fullname := deployments.GetFullName(d.Namespace, d.Name)
-			fmt.Printf("Linking '%s' to %s/%s...", dev.Mount.Source, fullname, dev.Swap.Deployment.Container)
+			fmt.Printf("Linking '%s' to %s/%s...", dev.Mount.Source, deployments.GetFullName(d.Namespace, d.Name), dev.Swap.Deployment.Container)
 			fmt.Println()
 			fmt.Printf("Ready! Go to your local IDE and continue coding!")
 			fmt.Println()
 
 			stopChannel := make(chan os.Signal, 1)
 			signal.Notify(stopChannel, os.Interrupt)
-			<-stopChannel
-			fmt.Println()
-			return nil
+
+			for {
+				select {
+				case <-stopChannel:
+					fmt.Println()
+					return nil
+				case <-reconnectChannel:
+					log.Infof("reconnecting")
+					shutdown(cancel, &wg)
+					storage.Stop(namespace, dev)
```

Comment:
I'm calling this because I've seen several times where shutdown will finish before calling storage.Stop, not sure if this is by design

Reply:
Fixed the root issue on a separate PR